grails-app/controllers/mini/asaas/PaymentController.groovy

```diff
@@ -0,0 +1,129 @@
+package mini.asaas
+
+import grails.validation.ValidationException
+
+import mini.asaas.BaseController
+import mini.asaas.enums.MessageType
+import mini.asaas.payment.Payment
+import mini.asaas.adapters.SavePaymentAdapter
+import mini.asaas.PaymentService
+
+
+class PaymentController extends BaseController {
+
+    PaymentService paymentService
+
+    def index() {
+        try {
+            List<Payment> payments = paymentService.list(params, getLimitPerPage(), getOffset())
+            return payments
+        } catch (Exception exception) {
+            String msg = "${message(code: "payment.index.error")}"
+            buildFlashAlert(msg, MessageType.ERROR, false)
+            redirect(url: '/dashboard')
+        }
+    }
+
+    def show() {
+        try {
+            Payment payment = paymentService.getById(params.id as Long, params.payerId as Long)
+            render(view: 'show', model: [payment: payment])
+        } catch (RuntimeException exception) {
+            buildFlashAlert(exception.message, MessageType.ERROR, false)
+            redirect(action: "index")
+        }
+    }
+
+    def create() {
+        try {
+            render(view: 'create')
+        } catch (Exception exception) {
+            String msg = "${message(code: "payment.create.error")}"
+            buildFlashAlert(msg, MessageType.ERROR, false)
+            redirect(action: "index")
+        }
+    }
+
+    def save() {
+        try {
+            SavePaymentAdapter adapter = new SavePaymentAdapter(params)
+            Payment payment = paymentService.create(adapter)
+            String flashMsg = "${message(code: "payment.created.success", args: [payment.id])}"
+            buildFlashAlert(flashMsg, MessageType.SUCCESS, true)
+            redirect(action: "show", id: payment.id)
+        } catch (ValidationException exception) {
+            String errMsg = "${message(code: "payment.created.error")}"
+            buildFlashAlert(errMsg, MessageType.ERROR, false)
+            render(view: "create", model: [adapter: adapter, errors: exception.errors])
+        } catch (RuntimeException exception) {
+            buildFlashAlert(exception.message, MessageType.ERROR, false)
+            render(view: "create", model: [adapter: adapter])
+        }
+    }
+
+    def edit() {
+        try {
+            Payment payment = paymentService.getById(params.id as Long, params.payerId as Long)
+            SavePaymentAdapter adapter = new SavePaymentAdapter(payment)
+            render(view: 'edit', model: [adapter: adapter])
```

Comment:
pq vc precisa de um adapter aqui? se isso aqui é só pra buscar os dados pra fazer a edição do payment, não é só vc buscar o payment e retornar ele? não precisa retornar adapter sempre, só se de fato vc for juntar dados e estrturuas que por si só não são associadas